src/cryptoadvance/specter/specter.py

```diff
@@ -52,7 +54,7 @@ def get_rpc(conf, old_rpc=None):
         rpc = BitcoinRPC(**conf)
     # check if we have something to compare with
     if old_rpc is None:
-        return rpc
+        return rpc if rpc.test_connection() else None
```

Comment:
```suggestion
        return rpc if rpc and rpc.test_connection() else None
```

Should first also check rpc is not not or you may get an error

Reply:
Good spots! As a result of this, i decided to do more tests especially on the cli_server side which is more customer-facing.